docs/plugins.md

```diff
@@ -1,3 +1,161 @@
+
+For example, `my-command <argument>` will require an argument to be provided and will throw a validation error otherwise. Alternatively, `my-command [argument]` will accept an argument, but will not throw when run without it. In that case, make sure to check for its presence.
+
+#### `func`
+
+Function that will be run when this command is executed. Receives an array of arguments, in order they were provided, a config object (see [`configuration` section](./configuration.md)) and options, that were passed to your command.
+
+You can return a Promise if your command is async.
+
+All errors are handled by the built-in logger. Prefer throwing instead of implementing your own logging mechanism.
+
+#### `options`
+
+An array of options that your command accepts.
+
+##### `options.name`
+
+Name of the option.
+
+For example, a `--reset-cache` option will result in a `resetCache: true` or `resetCache: false` present in the `options` object - passed to a command function as a last argument.
+
+Just like with a [command name](#name), your option can require a value (e.g. `--port <port>`) or accept an optional one (e.g. `--host [host]`). In this case, you may find [`default`](#optionsdefault) value useful.
+
+##### `options.description`
+
+Optional description of your option. When provided, will be used to output a better help information.
+
+##### `options.parse`
+
+Parsing function that can be used to transform a raw (string) option as passed by the user into a format expected by your function.
+
+##### `options.default`
+
+Default value for the option when not provided. Can be either a primitive value or a function, that receives a configuration and returns a primitive.
+
+Useful when you want to use project settings as default value for your option.
+
+#### `examples`
+
+An array of example usage of the command to be printed to the user.
+
+##### `examples.desc`
+
+String that describes this particular usage.
+
+##### `examples.cmd`
+
+A command with arguments and options (if applicable) that can be run in order to achieve the desired goal.
+
+## Migrating from `rnpm` configuration
+
+The changes are mostly cosmetic so the migration should be pretty straight-forward. 
+
+### Changing the configuration
+
+A `plugin` property should be renamed to `commands`.
+
+For example, the following `rnpm` configuration inside `package.json`:
+```json
+{
+  "rnpm": {
+    "plugin": "./path-to-commands.js",
+  }
+}
+```
+should be moved to a `react-native.config.js`:
+```js
+module.exports = {
+  commands: require('./path-to-commands.js')
+};
+```
+provided that `./path-to-commands.js` returns an array of commands.
+
+### Renaming command options
+
+If your command accepts options, rename `command` to `name`.
+
+For example,
+```js
+module.exports = {
+  name: 'foo',
+  func: () => console.log('My work'),
+  options: [
+    {
+      command: '--reset-cache, --resetCache',
+      description: 'Removes cached files',
+    }
+  ]
+}
+```
+should be changed to:
```

Comment:
I think it would be way easier to digest if we did a simple diff:

```diff
 module.exports = {
   name: 'foo',
   func: () => console.log('My work'),
   options: [
     {
-      command: '--reset-cache, --resetCache',
+      name: '--reset-cache, --resetCache',
       description: 'Removes cached files',
     }
   ]
 }
```